app/helpers/application_helper.rb

```diff
@@ -17,8 +19,18 @@ def md_summary(comment, word_count)
     simple_format(truncate(summary, length: word_count))
   end
 
-  def searchable_summary(comment, word_count)
+  def searchable_summary(comment, word_count, word = '')
     summary = strip_tags(md2html(comment)).gsub(/[\r\n]/, '')
-    truncate(summary, length: word_count)
+    words = word.split(/[[:space:]]+/).compact.reject(&:empty?) unless word.nil?
+    return truncate(summary, length: word_count) if words.blank?
+
+    words_pattern = words.map { |keyword| Regexp.escape(keyword) }.join('|')
+    words_regexp = Regexp.new(words_pattern, Regexp::IGNORECASE)
+    match = words_regexp.match(comment)
```

Comment:
```suggestion
    match = words_regexp.match(summary)
```

Markdownが取り除かれた`summary`が対象になるかと思います（34行目も同様）。
私のコメントのコードでややこしくしてしまったかもしれません😓